commands/apps/destroy.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+let co     = require('co');
+let cli    = require('heroku-cli-util');
+let extend = require('util')._extend;
+let _      = require('lodash');
+
+function* run (context, heroku) {
+  let git = require('../../lib/git')(context);
+
+  let app = context.args.app || context.app;
```

Comment:
@dickeyxxx should fail fast here if neither is passed, the get('/apps/') will succeed with a directory listing rather than giving some sort of not found message.